inference-engine/tests/functional/shared_test_classes/src/single_layer/matrix_nms.cpp

```diff
@@ -206,45 +255,43 @@ void MatrixNmsLayerTest::Compare(const std::vector<std::pair<ngraph::element::Ty
 
-    maxOutputBoxesPerClass = 0;
-    if (attrs.nms_top_k >= 0)
-        maxOutputBoxesPerClass = std::min(numBoxes, static_cast<size_t>(attrs.nms_top_k));
-    else
-        maxOutputBoxesPerClass = numBoxes;
+    init_input_shapes(shapes);
 
-    maxOutputBoxesPerBatch  = maxOutputBoxesPerClass * realClasses;
-    if (attrs.keep_top_k >= 0)
-        maxOutputBoxesPerBatch =
-                std::min(maxOutputBoxesPerBatch, static_cast<size_t>(attrs.keep_top_k));
-    Precision paramsPrec, maxBoxPrec, thrPrec;
+    ElementType paramsPrec, maxBoxPrec, thrPrec;
     std::tie(paramsPrec, maxBoxPrec, thrPrec) = inPrecisions;
-
-    const std::vector<size_t> boxesShape{numBatches, numBoxes, 4}, scoresShape{numBatches, numClasses, numBoxes};
-    auto ngPrc = convertIE2nGraphPrc(paramsPrec);
-    auto params = builder::makeParams(ngPrc, {boxesShape, scoresShape});
-    auto paramOuts = helpers::convert2OutputVector(helpers::castOps2Nodes<op::Parameter>(params));
-    auto nms = std::make_shared<opset8::MatrixNms>(paramOuts[0], paramOuts[1], attrs);
-    auto nms_0_identity = std::make_shared<opset5::Multiply>(nms->output(0), opset5::Constant::create(element::f32, Shape{1}, {1}));
-    auto nms_1_identity = std::make_shared<opset5::Multiply>(nms->output(1), opset5::Constant::create(attrs.output_type, Shape{1}, {1}));
-    auto nms_2_identity = std::make_shared<opset5::Multiply>(nms->output(2), opset5::Constant::create(attrs.output_type, Shape{1}, {1}));
-    function = std::make_shared<Function>(OutputVector{nms_0_identity, nms_1_identity, nms_2_identity}, params, "NMS");
+    const auto params = ngraph::builder::makeDynamicParams(paramsPrec, inputDynamicShapes);
+    const auto paramOuts =
+            ngraph::helpers::convert2OutputVector(ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(params));
+    auto nms = std::make_shared<opset8::MatrixNms>(paramOuts[0], paramOuts[1], m_attrs);
+    if (targetDevice == CommonTestUtils::DEVICE_CPU) {
+        m_outStaticShape = false;
+        auto result = std::make_shared<opset5::Result>(nms);
+        function = std::make_shared<Function>(result, params, "MatrixNMS");
+    } else {
+        m_outStaticShape = true;
+        auto nms_0_identity = std::make_shared<opset5::Multiply>(nms->output(0), opset5::Constant::create(element::f32, Shape{1}, {1}));
+        auto nms_1_identity = std::make_shared<opset5::Multiply>(nms->output(1), opset5::Constant::create(m_attrs.output_type, Shape{1}, {1}));
+        auto nms_2_identity = std::make_shared<opset5::Multiply>(nms->output(2), opset5::Constant::create(m_attrs.output_type, Shape{1}, {1}));
+        OutputVector results = {
+            std::make_shared<opset5::Result>(nms_0_identity),
+            std::make_shared<opset5::Result>(nms_1_identity),
+            std::make_shared<opset5::Result>(nms_2_identity)
+        };
+        function = std::make_shared<Function>(results, params, "MatrixNMS");
+    }
```

Comment:
```suggestion
    if (!m_outStaticShape) {
        auto result = std::make_shared<opset5::Result>(nms);
        function = std::make_shared<Function>(result, params, "MatrixNMS");
    } else {
        auto nms_0_identity = std::make_shared<opset5::Multiply>(nms->output(0), opset5::Constant::create(element::f32, Shape{1}, {1}));
        auto nms_1_identity = std::make_shared<opset5::Multiply>(nms->output(1), opset5::Constant::create(m_attrs.output_type, Shape{1}, {1}));
        auto nms_2_identity = std::make_shared<opset5::Multiply>(nms->output(2), opset5::Constant::create(m_attrs.output_type, Shape{1}, {1}));
        OutputVector results = {
            std::make_shared<opset5::Result>(nms_0_identity),
            std::make_shared<opset5::Result>(nms_1_identity),
            std::make_shared<opset5::Result>(nms_2_identity)
        };
        function = std::make_shared<Function>(results, params, "MatrixNMS");
    }
```

Reply:
Done. In order to cover the static/dynamic output shape test case I add the logic in the test code:
when testcase's input shape static output will be static shape too; if input shape dynamic, output will also be dynamic shape.